service/matching/version_sets.go

```diff
@@ -30,13 +30,25 @@ import (
 	"crypto/sha256"
 	"encoding/base64"
 
+	commonpb "go.temporal.io/api/common/v1"
 	"go.temporal.io/api/serviceerror"
 	"go.temporal.io/api/taskqueue/v1"
 	"go.temporal.io/api/workflowservice/v1"
 	"go.temporal.io/server/api/persistence/v1"
+	persistencespb "go.temporal.io/server/api/persistence/v1"
 	hlc "go.temporal.io/server/common/clock/hybrid_logical_clock"
 )
 
+var (
+	// TODO: go over error types, maybe not all should be invalid argument
+	errBuildNotFound                     = serviceerror.NewInvalidArgument("build id not found")
```

Comment:
```suggestion
	errBuildNotFound                     = serviceerror.NewInvalidArgument("build ID not found")
```

Reply:
Should be failed precondition and have the worker retry until an operator adds the build ID to the set.

Reply:
What if a task comes in with this build ID with no versioning information recorded (due to replication lag)?
Need to think this through but maybe we should let these polls through to the the hashed set ID generated by the build ID.

Reply:
Currently both codes are considered non-retryable.

I would use `OutOfRange` for now, since that is, until we eventually get error details that SDK reads to decide if specific errors should be allowed to be retryable.